### PR TITLE
2212 button toggle

### PR DESCRIPTION
**Pull request by AlexSofft**

### CODE REVIEW

**Comment by mf-code** [comment]
[5890] - K2Ke962a9qLp

**Comment by mf-code** [comment]
[5890] - PlS0A0MHMuEv

**Comment by mf-code** [comment]
[5890] - elFnMyHbSYTb

**Comment by mf-code** [comment]
[5890] - sUO6idilacce

**Comment by mf-code** [comment]
[5890] - QVz5YD8ljdoj

**Comment by mf-code** [comment]
[5890] - MD2ikIX8CuYZ

**Comment by mf-code** [comment]
[5890] - Mvz87DcVyghW

**Comment by mf-code** [comment]
[5890] - 6xYQS2esc3Sw

**Comment by mf-code** [comment]
[5890] - 8yHyZp1kBWYj

**Comment by mf-code** [comment]
[5890] - Fxd7ScVsLL8j

**Comment by mf-code** [comment]
[5890] - 2Q38iBP3xlMj

**Comment by mf-code** [comment]
[5890] - Pdm4kEaoWM4b

**Comment by mf-code** [comment]
[5890] - Ri219NAcz5zw

**Comment by mf-code** [comment]
[5890] - 7lMV438ohi7L

**Comment by mf-code** [comment]
[5890] - lyE09AfM1mrJ

**Comment by mf-code** [comment]
[5890] - N3K6xj4YyVWX

**Comment by mf-code** [comment]
[5890] - 40hNwE6hTpQC

**Comment by mf-code** [comment]
[5890] - 51MKDZIT7DzG

**Comment by mf-code** [comment]
[5890] - FZ1ElDDST7M6

**Comment by mf-code** [comment]
[5890] - RR7BMU0ZwPUQ

**Comment by mf-code** [comment]
[5890] - eM0fQCgnuuii

**Comment by mf-code** [comment]
[5890] - IuzdbhckGG6u

**Comment by mf-code** [comment]
[5890] - 7hDYw1T04Sb6

**Comment by mf-code** [comment]
[5890] - oIXUV8x06hi1

**Comment by mf-code** [comment]
[5890] - JG5yUWaOP2ks

**Comment by mf-code** [comment]
[5890] - 8K1tXQfnXByY

**Comment by mf-code** [comment]
[5890] - OFrmyAnuNyEA

**Comment by mf-code** [comment]
[5890] - openjdk8 - Allure report on Netlify: https://5f3d5333715f8a6a6f3b6321--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5890] - openjdk10 - Allure report on Netlify: https://5f3d552aee3f21d5b474e92a--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5898] - xktJIljmchS5

**Comment by mf-code** [comment]
[5898] - CPbZKlqvIMF9

**Comment by mf-code** [comment]
[5898] - dog9ZecOYEvI

**Comment by mf-code** [comment]
[5898] - dOgN7Q1xYKLs

**Comment by mf-code** [comment]
[5898] - NOqluHkFUjEs

**Comment by mf-code** [comment]
[5898] - P5O4ad52JH0J

**Comment by mf-code** [comment]
[5898] - d1YHDzUh6nEK

**Comment by mf-code** [comment]
[5898] - uJapgyysZZoD

**Comment by mf-code** [comment]
[5898] - lY8pDQDOt47T

**Comment by mf-code** [comment]
[5898] - LY7Zh1qfcqwK

**Comment by mf-code** [comment]
[5898] - GICb9nLfvJLZ

**Comment by mf-code** [comment]
[5898] - sl26gkBZbtVD

**Comment by mf-code** [comment]
[5898] - NhsXMurWJcUp

**Comment by mf-code** [comment]
[5899] - 4G09eyiAP6H1

**Comment by mf-code** [comment]
[5899] - ayuoJOxJEpW1

**Comment by mf-code** [comment]
[5899] - 9v0ntorZnOtO

**Comment by mf-code** [comment]
[5899] - WO5zDsZO2Wq4

**Comment by mf-code** [comment]
[5899] - ECFconZ2QK4O

**Comment by mf-code** [comment]
[5899] - ojma6bRZHWYt

**Comment by mf-code** [comment]
[5899] - q74IvAfAMgSr

**Comment by mf-code** [comment]
[5899] - C1c0QHD0NPXZ

**Comment by mf-code** [comment]
[5899] - GswbMNGh5v2e

**Comment by mf-code** [comment]
[5899] - sLEFfVYOBMUQ

**Comment by mf-code** [comment]
[5899] - WQSKnDnv31qk

**Comment by mf-code** [comment]
[5899] - nALbD8fPo0Nt

**Comment by mf-code** [comment]
[5908] - hbu7huyUqogS

**Comment by mf-code** [comment]
[5908] - PPicqdZZWe78

**Comment by mf-code** [comment]
[5908] - Uhh6memvysmf

**Comment by mf-code** [comment]
[5908] - GVXjypvLgMIy

**Comment by mf-code** [comment]
[5908] - q0BsbrdYwLLy

**Comment by mf-code** [comment]
[5908] - dkl8gUgqev4Q

**Comment by mf-code** [comment]
[5908] - 1br4poanfSIj

**Comment by mf-code** [comment]
[5908] - z1HljhnhUktk

**Comment by mf-code** [comment]
[5908] - LW7rGWRupfxH

**Comment by mf-code** [comment]
[5908] - e6fC4fA1MpwF

**Comment by mf-code** [comment]
[5908] - RBNKeSgXx63g

**Comment by mf-code** [comment]
[5908] - xYRojN0XBXiL

**Comment by mf-code** [comment]
[5908] - iFzONYI49oDr

**Comment by mf-code** [comment]
[5911] - W9hFYI6YtNck

**Comment by mf-code** [comment]
[5911] - qDlZsBfasXQ5

**Comment by mf-code** [comment]
[5911] - ziZD0MBaPTSC

**Comment by mf-code** [comment]
[5911] - zBFkdgEVCW9x

**Comment by mf-code** [comment]
[5911] - JL0gJuN8sZsZ

**Comment by mf-code** [comment]
[5911] - kzUa5rKjMCLI

**Comment by mf-code** [comment]
[5911] - JrhH3mGahPyg

**Comment by mf-code** [comment]
[5911] - P9dmkn4bvo33

**Comment by mf-code** [comment]
[5911] - BVfiSg8AYK5U

**Comment by mf-code** [comment]
[5911] - MFgGyt9Tu4qg

**Comment by mf-code** [comment]
[5911] - VMUL7vPjt2By

**Comment by mf-code** [comment]
[5911] - h8L1hu8PIah6

**Comment by mf-code** [comment]
[5912] - 5a8DGREjaw7s

**Comment by mf-code** [comment]
[5912] - yhlKpAKvz3gC

**Comment by mf-code** [comment]
[5912] - bx1BGqo5zZX2

**Comment by mf-code** [comment]
[5912] - zBYKTV6DWiJM

**Comment by mf-code** [comment]
[5911] - 6LOsib41TcnA

**Comment by mf-code** [comment]
[5911] - AeTsltinIPh8

**Comment by mf-code** [comment]
[5911] - openjdk8 - Allure report on Netlify: https://5f44e5024419c30116d96f62--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5911] - openjdk10 - Allure report on Netlify: https://5f44e5f7cc29f9019991b5fb--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5912] - TtBnPF2rorWW

**Comment by mf-code** [comment]
[5912] - lcl8rncHd3z3

**Comment by mf-code** [comment]
[5912] - uE0ZnyhOw718

**Comment by mf-code** [comment]
[5912] - SOryzj23guSi

**Comment by mf-code** [comment]
[5912] - ALGmGrTJ2iKB

**Comment by mf-code** [comment]
[5912] - PVlu0pAyv2yS

**Comment by mf-code** [comment]
[5912] - wHoIVm7ys5NQ

**Comment by mf-code** [comment]
[5912] - fxwvssuG8YN7

**Comment by mf-code** [comment]
[5912] - 7ZYth472YD38

**Comment by mf-code** [comment]
[5912] - GyqI7IFPNaIH

**Comment by mf-code** [comment]
[5912] - openjdk8 - Allure report on Netlify: https://5f44ebad46b7dd6396c0c065--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5912] - openjdk10 - Allure report on Netlify: https://5f44ecdacc29f9870d91b79b--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5916] - JcXhRBREo9XY

**Comment by mf-code** [comment]
[5916] - sQjzpCVbhgrw

**Comment by mf-code** [comment]
[5916] - oIz4QLfmZtbn

**Comment by mf-code** [comment]
[5916] - Io64cXo7h4Ej

**Comment by mf-code** [comment]
[5916] - tk2hOdt2bnPF

**Comment by mf-code** [comment]
[5916] - jFrfzL521ghw

**Comment by mf-code** [comment]
[5916] - FH7pDgMJVYnZ

**Comment by mf-code** [comment]
[5916] - QCvBDkihxuU4

**Comment by mf-code** [comment]
[5916] - bhb5cG91p2oa

**Comment by mf-code** [comment]
[5916] - kPi2rPcNoMh0

**Comment by mf-code** [comment]
[5916] - XWBPgp7dxHkT

**Comment by mf-code** [comment]
[5916] - guvCKfq5aUEf

**Comment by mf-code** [comment]
[5916] - byU2EjC4QhKn

**Comment by mf-code** [comment]
[5916] - sVJn3t2zseIE

**Comment by mf-code** [comment]
[5916] - openjdk8 - Allure report on Netlify: https://5f451ec646b7ddd0d7c0c6fd--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5916] - openjdk10 - Allure report on Netlify: https://5f451ff405afac240ff44f01--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5922] - 6UEsQ3qSTNVz

**Comment by mf-code** [comment]
[5922] - aabzEQGM3zZk

**Comment by mf-code** [comment]
[5922] - xmDpDzc5pWj7

**Comment by mf-code** [comment]
[5922] - PPdS9ZecQP0O

**Comment by mf-code** [comment]
[5922] - FdVqSLY9n8u2

**Comment by mf-code** [comment]
[5922] - 8vq9PucVWia2

**Comment by mf-code** [comment]
[5922] - 7R3xHx1Exd6v

**Comment by mf-code** [comment]
[5922] - 4a3zn3WSJupx

**Comment by mf-code** [comment]
[5922] - e7o9Aromxwus

**Comment by mf-code** [comment]
[5922] - VsW5lmbyAWCu

**Comment by mf-code** [comment]
[5922] - bX1l3Xq9bhF2

**Comment by mf-code** [comment]
[5922] - CxNsW8j9Klf0

**Comment by mf-code** [comment]
[5925] - Ywc6RJNmlBI6

**Comment by mf-code** [comment]
[5925] - cgwk0rk7bv70

**Comment by mf-code** [comment]
[5925] - NsjHUVE70wjY

**Comment by mf-code** [comment]
[5925] - 80xJtN9TAlti

**Comment by mf-code** [comment]
[5925] - dmjyDPsAKsGi

**Comment by mf-code** [comment]
[5925] - fzPv9rY4ecLu

**Comment by mf-code** [comment]
[5925] - vgdtnIVTgWU4

**Comment by mf-code** [comment]
[5925] - Pl2XiFUKD9he

**Comment by mf-code** [comment]
[5925] - tH9kRXosVoru

**Comment by mf-code** [comment]
[5925] - ZeWM3K4U88c1

**Comment by mf-code** [comment]
[5925] - gdbiPZEDIlY0

**Comment by mf-code** [comment]
[5925] - yfW7VpTSHHCf

**Comment by mf-code** [comment]
[5925] - CK5Eh8DVF7T5

**Comment by mf-code** [comment]
[5925] - ofkmdHIIxnaV

**Comment by mf-code** [comment]
[5925] - 7mL1JVfyNdOX

**Comment by mf-code** [comment]
[5925] - DPEqmqx1o5FU

**Comment by mf-code** [comment]
[5925] - W9SgXo4s2T85

**Comment by mf-code** [comment]
[5925] - XsXz3G0uRwOM

**Comment by mf-code** [comment]
[5925] - fH8G6hmJFb1n

**Comment by mf-code** [comment]
[5925] - qYR77DEOo7FW

**Comment by mf-code** [comment]
[5925] - 1SsnZqhW1upe

**Comment by mf-code** [comment]
[5925] - LzMHFWEgCjcb

**Comment by mf-code** [comment]
[5925] - DK7RzNetttht

**Comment by mf-code** [comment]
[5925] - Nl4hF1uCdjwL

**Comment by mf-code** [comment]
[5925] - cQSX4bVwnh2x

**Comment by mf-code** [comment]
[5925] - fqKmJCOrbiHM

**Comment by mf-code** [comment]
[5925] - nldxKRGmG9YY

**Comment by mf-code** [comment]
[5925] - H5dtTikPLmND

**Comment by mf-code** [comment]
[5925] - QtOqTtQq3YKk

**Comment by mf-code** [comment]
[5925] - Xiapct6PHani

**Comment by mf-code** [comment]
[5925] - 9Q8I88qKqBn7

**Comment by mf-code** [comment]
[5925] - daKfkRPDwpke

**Comment by mf-code** [comment]
[5925] - pq9viEOsD8Ox

**Comment by mf-code** [comment]
[5925] - SEADerXe2cjO

**Comment by mf-code** [comment]
[5925] - N0fmFKPSBZl4

**Comment by mf-code** [comment]
[5925] - XRiQMSvh3Yq9

**Comment by mf-code** [comment]
[5925] - ZiMPzKMlTc04

**Comment by mf-code** [comment]
[5925] - aWvBAI2lQNMo

**Comment by mf-code** [comment]
[5942] - w6Ys2lBR0h9r

**Comment by mf-code** [comment]
[5942] - ow3B4maCHEcH

**Comment by mf-code** [comment]
[5942] - aW85Fj6SVgGZ

**Comment by mf-code** [comment]
[5942] - h0cqbZEFwQG4

**Comment by mf-code** [comment]
[5942] - 8jRoEeqGaqBM

**Comment by mf-code** [comment]
[5942] - OEUvhukAxCyM

**Comment by mf-code** [comment]
[5942] - AesSoySCKC9v

**Comment by mf-code** [comment]
[5942] - rUO6NGB8yhxT

**Comment by mf-code** [comment]
[5942] - zwoXrpNNpFQA

**Comment by mf-code** [comment]
[5942] - iGp0f9GLx8Wu

**Comment by mf-code** [comment]
[5942] - Jy2uNOm5TPDe

**Comment by mf-code** [comment]
[5942] - E4KCmpsBjs4t

**Comment by mf-code** [comment]
[5942] - VXm2E1I41Mu3

**Comment by mf-code** [comment]
[5942] - 9Z4JqrZX5gV1

**Comment by mf-code** [comment]
[5942] - bchxtjzr4TRD

**Comment by mf-code** [comment]
[5942] - THTf6tkOY2H6

**Comment by mf-code** [comment]
[5942] - openjdk8 - Allure report on Netlify: https://5f48b43a4801d5ed3a683fe5--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5942] - openjdk10 - Allure report on Netlify: https://5f48b4f824ed6498e2ac5d38--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5942] - XslXLTi9wTmY

**Comment by mf-code** [comment]
[5942] - uZPjbGm4nHyu

**Comment by mf-code** [comment]
[5942] - 1uClSC9eIm5d

**Comment by mf-code** [comment]
[5942] - oXbFsBk4h2Mg

**Comment by mf-code** [comment]
[5942] - 4HKnexYCGWI7

**Comment by mf-code** [comment]
[5942] - z34oXkVEcBNc

**Comment by mf-code** [comment]
[5942] - QguwSwjECezm

**Comment by mf-code** [comment]
[5942] - P7iQf61a2UKq

**Comment by mf-code** [comment]
[5942] - o7MTPv2KGlPi

**Comment by mf-code** [comment]
[5942] - Bta04HDsmqsO

**Comment by mf-code** [comment]
[5942] - NkFTQzihRSFJ

**Comment by mf-code** [comment]
[5942] - 1fdcBEVkMw9U

**Comment by mf-code** [comment]
[5942] - 6FurvwCF7PJ2

**Comment by mf-code** [comment]
[5942] - WQgeOvW5kIm5

**Comment by mf-code** [comment]
[5942] - cSaaF5sOTrMW

**Comment by mf-code** [comment]
[5942] - 7IItpEtupXhS

**Comment by mf-code** [comment]
[5942] - openjdk8 - Allure report on Netlify: https://5f48e4e50664ec01a861d3ca--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5942] - openjdk10 - Allure report on Netlify: https://5f48e60e945777015679a005--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5951] - fF2mHTFVH5r5

**Comment by mf-code** [comment]
[5951] - C6GPwXu5Mzb3

**Comment by mf-code** [comment]
[5951] - Zrjc4EyP6Yu8

**Comment by mf-code** [comment]
[5951] - aBrIvIsWG8wk

**Comment by mf-code** [comment]
[5951] - PHqUISi3RKfU

**Comment by mf-code** [comment]
[5951] - X2w7fPkpHA9h

**Comment by mf-code** [comment]
[5951] - gBqhS1NA4ktT

**Comment by mf-code** [comment]
[5951] - ok9c1EV8YqV4

**Comment by mf-code** [comment]
[5951] - 9f71S9LQjCZl

**Comment by mf-code** [comment]
[5951] - 9HCVIhW2jQsS

**Comment by mf-code** [comment]
[5951] - neW2m9h5p4jG

**Comment by mf-code** [comment]
[5951] - e9f8BVqSEK19

**Comment by mf-code** [comment]
[5951] - ktly9p7Ztj7I

**Comment by mf-code** [comment]
[5951] - ydmty3ZaPXXg

**Comment by mf-code** [comment]
[5951] - d6PIxQqHEkRN

**Comment by mf-code** [comment]
[5960] - dCjmSoJLnnFD

**Comment by mf-code** [comment]
[5960] - 8BIwoqdGw7oA

**Comment by mf-code** [comment]
[5960] - hgDtpUyE8fvx

**Comment by mf-code** [comment]
[5960] - FwkiVJl8DhL1

**Comment by mf-code** [comment]
[5960] - gepax9RKTn2G

**Comment by mf-code** [comment]
[5960] - MtuDEcNMxc4D

**Comment by mf-code** [comment]
[5960] - Yvb2mXDyN9XV

**Comment by mf-code** [comment]
[5960] - nnrHDX5f73Jj

**Comment by mf-code** [comment]
[5960] - oY8dgdPvlviT

**Comment by mf-code** [comment]
[5960] - lc18N2mEzCq6

**Comment by mf-code** [comment]
[5960] - EoUWOV9k7jao

**Comment by mf-code** [comment]
[5960] - jnNAXI8wdLFf

**Comment by mf-code** [comment]
[5960] - f2odpxLrd56l

**Comment by mf-code** [comment]
[5960] - 0lrKpQw4IMJO

**Comment by mf-code** [comment]
[5960] - J0NsrrCzxpYl

**Comment by mf-code** [comment]
[5960] - CDTraFBdaC6F

**Comment by mf-code** [comment]
[5960] - ulyZUaSOMPLf

**Comment by mf-code** [comment]
[5960] - z3pnp42lwzaN

**Comment by mf-code** [comment]
[5960] - openjdk8 - Allure report on Netlify: https://5f491ed90911a152a538a299--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5960] - openjdk10 - Allure report on Netlify: https://5f492077bd26990162d59d58--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5964] - zYTID454GHZh

**Comment by mf-code** [comment]
[5964] - o3H6EqYNGreT

**Comment by mf-code** [comment]
[5964] - DWXcmsaKzILf

**Comment by mf-code** [comment]
[5964] - uLc7OZewIlUU

**Comment by mf-code** [comment]
[5964] - MSxAttrZ0oU9

**Comment by mf-code** [comment]
[5964] - dSDbGymoRvcM

**Comment by mf-code** [comment]
[5964] - QW5lO3CR2VOO

**Comment by mf-code** [comment]
[5964] - IcdUpAGeD2vY

**Comment by mf-code** [comment]
[5964] - EBWt2md3iJZI

**Comment by mf-code** [comment]
[5964] - 4jQkjkGQflrW

**Comment by mf-code** [comment]
[5964] - U9Vvy8sRliSC

**Comment by mf-code** [comment]
[5964] - jREhMQ83lVVU

**Comment by mf-code** [comment]
[5964] - emUupdz5jId6

**Comment by mf-code** [comment]
[5964] - CpBsKQZaXBzJ

**Comment by mf-code** [comment]
[5964] - MH8ibpYZweSL

**Comment by mf-code** [comment]
[5964] - Ycq8naDuxWsk

**Comment by mf-code** [comment]
[5964] - czl2px9NUbFS

**Comment by mf-code** [comment]
[5964] - oKjwiQIRSidc

**Comment by mf-code** [comment]
[5964] - openjdk8 - Allure report on Netlify: https://5f4a1618cedffd0df861a91e--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5964] - openjdk10 - Allure report on Netlify: https://5f4a17c086e53734d1edaec2--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5964] - 9r22UikvhqI3

**Comment by mf-code** [comment]
[5964] - leswAWlMhLyY

**Comment by mf-code** [comment]
[5964] - FdPYXauB8P6v

**Comment by mf-code** [comment]
[5964] -

**Comment by mf-code** [comment]
[5964] - cbTdm0w5rX3B

**Comment by mf-code** [comment]
[5964] - peFTWWKwcPPB

**Comment by mf-code** [comment]
[5964] - 5qtWRJbodxFN

**Comment by mf-code** [comment]
[5964] - Gwmn7ukedvtE

**Comment by mf-code** [comment]
[5964] - nHmmUSsoy6bH

**Comment by mf-code** [comment]
[5964] - FSVKnYcb04Qu

**Comment by mf-code** [comment]
[5964] - URBe72bmOyCn

**Comment by mf-code** [comment]
[5964] - l5IILSKSgl5q

**Comment by mf-code** [comment]
[5964] - kAkOgy3NRcNu

**Comment by mf-code** [comment]
[5964] - 86JDguexfVnu

**Comment by mf-code** [comment]
[5964] - zgYJRzu9CAyS

**Comment by mf-code** [comment]
[5964] - uoC9FKjNku7R

**Comment by mf-code** [comment]
[5964] - 5UXmaSTrXFI6

**Comment by mf-code** [comment]
[5964] - VqgpXhIIkSmS

**Comment by mf-code** [comment]
[5964] - openjdk8 - Allure report on Netlify: https://5f4a84b483248da4fef6b9e1--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5964] - openjdk10 - Allure report on Netlify: https://5f4a87145bf93094e7ad3363--musing-swartz-6fc060.netlify.app

**Review by AkioFaris — APPROVED** [review]

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ButtonToggleTests.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void verifyButtonToggle() {
+        basicButtonToggle.is().displayed();
+        basicButtonToggle.is().enabled();
+        basicButtonToggle.clickButtonToggleByValue(ITALIC);
+        basicButtonToggle.is().assertButtonToggleIsSelected(ITALIC);
+        basicButtonToggle.is().assertButtonToggleButtonIsPressed(ITALIC);
+        basicButtonToggle.is().assertButtonToggleButtonHasText(ITALIC);
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ButtonToggleTests.java:33]: It's too much checks in one test. Could you please split them into several parts.

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ButtonToggleTests.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+public class ButtonToggleTests extends TestsInit {
+
+    private final String ITALIC = "italic";
+    private final String BOLD = "bold";
+    private final String UNDERLINE = "underline";
+    private final String LEFT = "left";
+    private final String CENTER = "center";
+    private final String RIGHT = "right";
+
+    @BeforeMethod
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ButtonToggleTests.java:20]: Are you sure that method always runs? From my perspective the parameter "always run = true" is required here.

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ButtonToggleTests.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static io.github.com.StaticSite.angularPage;
+import static io.github.com.pages.sections.ButtonToggleSection.*;
+import static io.github.epam.site.steps.States.shouldBeLoggedIn;
+
+public class ButtonToggleTests extends TestsInit {
+
+    private final String ITALIC = "italic";
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ButtonToggleTests.java:13]: Code convention convention not followed here.

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ButtonToggleTests.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void verifyButtonToggleCombineSelection() {
+        basicButtonToggle.clickButtonToggleByValue(BOLD);
+        basicButtonToggle.clickButtonToggleByValue(UNDERLINE);
+        basicButtonToggle.is().assertButtonToggleIsSelected(BOLD);
+        basicButtonToggle.is().assertButtonToggleIsSelected(UNDERLINE);
+        basicButtonToggle.clickButtonToggleByValue(BOLD);
+        basicButtonToggle.is().assertButtonToggleIsNotSelected(BOLD);
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ButtonToggleTests.java:43]: the same

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ButtonToggleTests.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void verifyExclusiveButtonToggle() {
+        basicButtonToggleAlign.is().displayed();
+        basicButtonToggleAlign.is().enabled();
+        basicButtonToggleAlign.clickButtonToggleByValue(LEFT);
+        basicButtonToggleAlign.is().assertButtonToggleIsSelected(LEFT);
+        selectedValue.has().text("Selected value: left");
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/ButtonToggleTests.java:52]: the same

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/unit/ButtonToggleUnitTests.java [hunk 6]
```diff
@@ -5,14 +5,17 @@
...
+import static io.github.com.pages.sections.ButtonToggleSection.basicButtonToggleAlign;
 import static io.github.epam.site.steps.States.shouldBeLoggedIn;
+import static org.testng.Assert.assertTrue;
 
 public class ButtonToggleUnitTests extends TestsInit {
 
-    public static final String CLASS_ATTR = "class";
-    public static final String ACCENT = "accent";
-    public static final String PRIMARY = "primary";
+    private final String ITALIC = "italic";
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/unit/ButtonToggleUnitTests.java:15]: Code convention convention not followed here.

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/ButtonToggleAssert.java [hunk 7]
```diff
@@ -2,7 +2,34 @@
...
 import com.epam.jdi.light.asserts.generic.UIAssert;
+import com.epam.jdi.light.common.JDIAction;
+import org.hamcrest.Matchers;
+
+import static com.epam.jdi.light.asserts.core.SoftAssert.jdiAssert;
 
 public class ButtonToggleAssert extends UIAssert<ButtonToggleAssert, ButtonToggle> {
 
+    @JDIAction("Assert that '{name}' has '{0}' class")
+    public ButtonToggleAssert assertButtonToggleIsSelected(String value) {
```
Comment (gbuh) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/ButtonToggleAssert.java:13]: Assertion method name is not reasonable. From my point of view "selected" will be fairly well here.

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/ButtonToggleAssert.java [hunk 8]
```diff
@@ -2,7 +2,34 @@
...
+        jdiAssert(element.isButtonToggleSelected(value), Matchers.is(true));
+        return this;
+    }
+
+    @JDIAction("Assert that '{name}' has '{0}' class")
+    public ButtonToggleAssert assertButtonToggleIsNotSelected(String value) {
```
Comment (gbuh) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/ButtonToggleAssert.java:19]: To my mined, this method is redundant.

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/ButtonToggleAssert.java [hunk 9]
```diff
@@ -2,7 +2,34 @@
...
+        jdiAssert(element.isButtonToggleSelected(value), Matchers.is(false));
+        return this;
+    }
+
+    @JDIAction("Assert that '{name}' has attr '{0}'")
+    public ButtonToggleAssert assertButtonToggleButtonIsPressed(String value) {
```
Comment (gbuh) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/ButtonToggleAssert.java:25]: Assertion method name is not reasonable. From my point of view, the "pressed" will be fine here.

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/ButtonToggleAssert.java [hunk 10]
```diff
@@ -2,7 +2,34 @@
...
+        jdiAssert(element.isButtonToggleButtonPressed(value), Matchers.is(true));
+        return this;
+    }
+
+    @JDIAction("Assert that '{name}' has text '{0}'")
+    public ButtonToggleAssert assertButtonToggleButtonHasText(String value) {
```
Comment (gbuh) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/ButtonToggleAssert.java:31]: Assertion method name is not reasonable. From my point of view, the "text" will be fine here.

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/ButtonToggle.java [hunk 11]
```diff
@@ -5,26 +5,56 @@
...
 import com.epam.jdi.light.elements.common.UIElement;
 import com.epam.jdi.light.elements.complex.WebList;
+import org.openqa.selenium.By;
 
 public class ButtonToggle extends UIBaseElement<ButtonToggleAssert> {
 
     @JDIAction("Click '{name}' button by tag value '{0}'")
-    public void clickButtonByValue(String value){
-        getButtonByTagValue(value).click();
+    public void clickButtonToggleByValue(String value) {
```
Comment (gbuh) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/ButtonToggle.java:13]: Assertion method name is not reasonable. From my point of view, the "click" will be fine here.

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/ButtonToggle.java [hunk 12]
```diff
@@ -5,26 +5,56 @@
...
+        getButtonToggleByTagValue(value).click();
+    }
+
+    @JDIAction("Is '{name}' pressed '{0}'")
+    public boolean isButtonToggleButtonPressed(String value) {
```
Comment (gbuh) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/ButtonToggle.java:18]: Assertion method name is not reasonable. From my point of view, the "isPressed" will be fine here.

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/ButtonToggle.java [hunk 13]
```diff
@@ -5,26 +5,56 @@
...
+        return getButtonToggleByTagValue(value).find(By.cssSelector("button")).attr("aria-pressed").equals("true");
+    }
+
+    @JDIAction("'{name}' has text '{0}'")
+    public boolean buttonToggleHasText(String value) {
+        return getButtonToggleByTagValue(value).find(By.cssSelector("button")).getText().equalsIgnoreCase(value);
     }
 
     @JDIAction("Get '{name}' tabs")
```
Comment (gbuh) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/ButtonToggle.java:27]: Annotation @JDIAction("Get '{name}' tabs") is not required here. Method is private.

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/ButtonToggle.java [hunk 14]
```diff
@@ -5,26 +5,56 @@
...
-    private WebList getButtonToggleGroups() {
+    private WebList getButtonToggleItems() {
         return this.finds(".mat-button-toggle");
     }
 
-    private UIElement getButtonByTagValue(String value) {
+    @JDIAction("Is '{name}' selected '{0}'")
+    public boolean isButtonToggleSelected(String value) {
```
Comment (gbuh) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/ButtonToggle.java:33]: Assertion method name is not reasonable. From my point of view, the "isSelected" will be fine here.

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/ButtonToggle.java [hunk 15]
```diff
@@ -5,26 +5,56 @@
...
+        UIElement element = getButtonToggleByTagValue(value);
+        if (isButtonToggleDisabled(value)) {
+            throw new UnsupportedOperationException(String.format("button toggle is disabled %s", value));
+        }
+        return element.attr("class").contains("checked");
+    }
+
+    private boolean isButtonToggleDisabled(String value) {
```
Comment (gbuh) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/ButtonToggle.java:41]: Assertion method name is not reasonable. From my point of view, the "isDisabled" will be fine here.

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/ButtonToggle.java [hunk 16]
```diff
@@ -5,26 +5,56 @@
...
+        for (UIElement e : getButtonToggleItems()) {
             if (e.attr("value").equals(value)) {
                 element = e;
             }
         }
         return element;
     }
+
+    @Override
+    public ButtonToggleAssert is() {
```
Comment (gbuh) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/ButtonToggle.java:57]: First there should be "public" methods, then "protected", then "private".

**Pull request merged**